projects/tests/test_create_delete_environments.py

```diff
@@ -0,0 +1,129 @@
+from django.contrib.auth import get_user_model
+from django.contrib.messages import get_messages
+from django.test import TestCase
+
+from apps.models import Apps, JupyterInstance, Subdomain
+from projects.models import Environment, Project
+from projects.views import can_model_instance_be_deleted
+
+User = get_user_model()
+
+test_user = {"username": "foo1", "email": "foo@test.com", "password": "bar"}
+test_superuser = {"username": "superuser", "email": "superuser@test.com", "password": "bar"}
+
+
+class EnvironmentTestCase(TestCase):
+    def setUp(self):
+        user = User.objects.create_user(test_user["username"], test_user["email"], test_user["password"])
+        project = Project.objects.create_project(name="test-env", owner=user, description="")
+        User.objects.create_superuser(test_superuser["username"], test_superuser["email"], test_superuser["password"])
+        app = Apps.objects.create(name="Some App", slug="someapp")
+        Environment.objects.create(app=app, name="env-to-be-deleted", project=project)
+
+    def test_environment_creation_user(self):
+        """
+        Test regular user not allowed to create environment
+        """
+        self.client.login(username=test_user["email"], password=test_user["password"])
+
+        project = Project.objects.get(name="test-env")
+        app = Apps.objects.get(slug="someapp")
+
+        n_envs_before = len(Environment.objects.all())
+        response = self.client.post(
+            f"/projects/{project.slug}/createenvironment/",
+            {
+                "environment_name": "new-env-user",
+                "environment_repository": "n",
+                "environment_image": "n",
+                "environment_app": "n",
+                "app": app.pk,
+            },
+        )
+        self.assertEqual(response.status_code, 403)
+        n_envs_after = len(Environment.objects.all())
+
+        self.assertEqual(n_envs_before, n_envs_after)
+
+        """
+        Test regular user not allowed to delete environment
+        """
```

Comment:
```suggestion
        # Test regular user not allowed to delete environment
```

Or could this test be split into two?

Reply:
And the ones bellow as well?

Reply:
What is the benefit of splitting them into two tests?

Reply:
They just seemed like a two connected tests, that could be easily separated. And I think the smaller a unittest the better